### DISCUSSION TITLE
Some Tooltips segments not working

### DISCUSSION THREAD

**Discussion by spavett** [JanDeDobbeleer/oh-my-posh#3243] [Troubleshoot] [answered]
I've customised some tooltips into my profile as so but only az and kubectl work. Nothing shows up for the others. Any ideas?

"tooltips": [
        {
            "foreground": "#ffffff",
            "background": "#7FD5EA",
            "style": "diamond",
            "leading_diamond": "\ue0b6",
            "trailing_diamond": "\ue0b4",
            "template": " \u202D\uFCD1 {{ .Full }} ",
            "type": "go",
            "tips": ["go"]
        },
        {
            "background": "#ebcc34",
            "foreground": "#000000",
            "style": "diamond",
            "leading_diamond": "\ue0b6",
            "trailing_diamond": "\ue0b4",
            "template": " \ufd31 Kubernetes {{.Context}} cluster {{if .Namespace}}- {{.Namespace}} namespace{{end}} ",
            "type": "kubectl",
            "tips": ["kubectl"]
        },
        {
            "background": "#9ec3f0",
            "foreground": "#000000",
            "style": "diamond",
            "leading_diamond": "\ue0b6",
            "trailing_diamond": "\ue0b4",
            "template": " \ufd03 Subscription {{ .Name }} ({{ if .EnvironmentName | contains \"AzureCloud\" }}{{ \"Global\" }}{{ else }}{{ .EnvironmentName }}{{ end }}) ",
            "type": "az",
            "tips": ["az"]
        },
        {
            "type": "node",
            "tips": ["node"],
            "style": "diamond",
            "leading_diamond": "\ue0b6",
            "trailing_diamond": "\ue0b4",
            "foreground": "#ffffff",
            "background": "#6CA35E",
            "template": " \uE718 {{ .Full }} "
        },
        {
            "type": "rust",
            "tips": ["rustc"],
            "style": "diamond",
            "leading_diamond": "\ue0b6",
            "trailing_diamond": "\ue0b4",
            "foreground": "#193549",
            "background": "#99908a",
            "template": " \uE7a8 {{ .Full }} "
        }
    ],

**Comment by JanDeDobbeleer** [comment, marked as answer]
@spavett probably because those are language segments and the default `display_mode` is `context` and not `always`. Changing it to that would enable it for tooltips, but I bet you also need to enable fetching the version as otherwise there's no additional information other than the language icon.

**Reply by spavett** [reply]
Thank @JanDeDobbeleer. I tried the following but it didn't help.  

    "tooltips": [
        {
            "foreground": "#ffffff",
            "background": "#7FD5EA",
            "style": "diamond",
            "leading_diamond": "\ue0b6",
            "trailing_diamond": "\ue0b4",
            "template": " \u202D\uFCD1 {{ .Full }} ",
            "type": "go",
            "tips": ["go"],
            "display_mode": "always",
            "fetch_version": true
        },
        {
            "background": "#ebcc34",
            "foreground": "#000000",
            "style": "diamond",
            "leading_diamond": "\ue0b6",
            "trailing_diamond": "\ue0b4",
            "template": " \ufd31 Kubernetes {{.Context}} cluster {{if .Namespace}}- {{.Namespace}} namespace{{end}} ",
            "type": "kubectl",
            "tips": ["kubectl"]
        },
        {
            "background": "#9ec3f0",
            "foreground": "#000000",
            "style": "diamond",
            "leading_diamond": "\ue0b6",
            "trailing_diamond": "\ue0b4",
            "template": " \ufd03 Subscription {{ .Name }} ({{ if .EnvironmentName | contains \"AzureCloud\" }}{{ \"Global\" }}{{ else }}{{ .EnvironmentName }}{{ end }}) ",
            "type": "az",
            "tips": ["az"]
        },
        {
            "type": "node",
            "tips": ["node"],
            "display_mode": "always",
            "fetch_version": true,
            "missing_command_text": "node is missing",
            "style": "diamond",
            "leading_diamond": "\ue0b6",
            "trailing_diamond": "\ue0b4",
            "foreground": "#ffffff",
            "background": "#6CA35E",
            "template": " \uE718 {{ if .PackageManagerIcon }}{{ .PackageManagerIcon }} {{ end }}{{ .Full }} "
        },
        {
            "type": "python",
            "tips": ["python"],
            "display_mode": "always",
            "fetch_version": true,
            "style": "diamond",
            "leading_diamond": "\ue0b6",
            "trailing_diamond": "\ue0b4",
            "foreground": "#100e23",
            "background": "#906cff",
            "template": " \uE235 {{ .Full }} "
        },
        {
            "type": "rust",
            "tips": ["rustc"],
            "display_mode": "always",
            "fetch_version": true,
            "style": "diamond",
            "leading_diamond": "\ue0b6",
            "trailing_diamond": "\ue0b4",
            "foreground": "#193549",
            "background": "#99908a",
            "template": " \uE7a8 {{ .Full }} "
        }
    ]

**Reply by JanDeDobbeleer** [reply]
@spavett `display_mode` needs to go inside the `properties` block. See the [git](https://ohmyposh.dev/docs/segments/git#sample-configuration) segment sample for example.

**Reply by spavett** [reply]
That works (along with the "home_enabled": true - as it won't show in ~ e.g.

        { 
            "type": "rust",
            "tips": ["rustc"],
            "properties": {
                "home_enabled": true,
                "display_mode": "always",
                "fetch_version": true
            },
            "style": "diamond",
            "leading_diamond": "\ue0b6",
            "trailing_diamond": "\ue0b4",
            "foreground": "#193549",
            "background": "#99908a",
            "template": " \uE7a8 {{ .Full }} "
        }
